metadata/metadata.go

```diff
@@ -219,12 +220,32 @@ func (err *ResourceExists) GRPCStatus() *status.Status {
 	return status.New(codes.AlreadyExists, err.Error())
 }
 
+type ResourceChanged struct {
+	ID ResourceID
+}
+
+func (err *ResourceChanged) Error() string {
+	id := err.ID
+	errMsg := fmt.Sprintf("%s changed. Use a new variant.", id.String())
```

Comment:
What does it mean for a resource to change and why use a new variant?

Reply:
I might just move this part to another PR. The idea is if you try and register a variant that already exists, if the key attributes on it have changed (for example for a feature, the type has changed) it should say; "look dude, you changed some important shit on this variant; you should register it as a new variant."

Reply:
Is this if you manually set a variant then change it?

Reply:
yisshhh